### PR TITLE
don't allow installation of already installed apps - fixes #14004

### LINKED ISSUE #14004: App installer does not reflect installed apps.
Hello,

In my Owncloud 8 instance, I have discovered that after a page refresh, the app installation interface does not "remember" which apps are installed, allowing me to attempt to install an already installed app, and failing to do so.

In the below example, I have installed Contacts and Calendar 8.0 via the installation interface, then tested them, then came back to the admin page to check for further apps to install, and I found that I was being prompted to install the packages again. When I clicked the "Enable" button, it sat there for about 20-30 seconds, before providing an error, "App Directory Already Exists".

http://i.imgur.com/26y2ytV.png   Here is a screenshot of this, with the owncloud menu opened to show that the apps are indeed already installed, as well as the error message mentioned above.

http://i.imgur.com/9vGeEF0.png  And here is a screenshot of the apps installed in the "Enabled" section of the app interface, to show that they are indeed installed.

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #14004 for master aka 8.1

holds class renaming of OC_OCSClient -> \OC\OCSClient

@karlitschek @PVince81 please review - THX

### CODE REVIEW

File: lib/private/httphelper.php [hunk 1]
```diff
@@ -70,6 +70,7 @@ public function getUrlContent($url) {
 			curl_setopt($curl, CURLOPT_URL, $url);
 			curl_setopt($curl, CURLOPT_PROTOCOLS,  CURLPROTO_HTTP | CURLPROTO_HTTPS);
 			curl_setopt($curl, CURLOPT_REDIR_PROTOCOLS,  CURLPROTO_HTTP | CURLPROTO_HTTPS);
+			curl_setopt($curl, CURLOPT_SSL_VERIFYPEER,  0);
```
Comment (DeepDiver1975) [inline lib/private/httphelper.php]: grrrr - bloody debug code

**Comment by LukasReschke** [comment]
@owncloud-bot Retest this please.

**Comment by DeepDiver1975** [comment]
@owncloud-bot Retest this please.

File: lib/private/ocsclient.php [hunk 2]
```diff
@@ -42,53 +47,44 @@ protected static function isAppstoreEnabled() {
...
-		if(!self::isAppstoreEnabled()) {
+		if (!self::isAppStoreEnabled()) {
 			return null;
 		}
-		$url=OC_OCSClient::getAppStoreURL().'/content/categories';
-		$xml=OC_OCSClient::getOCSresponse($url);
-		if($xml==false) {
+		$url = self::getAppStoreURL() . '/content/categories';
+		$xml = \OC::$server->getHTTPHelper()->getUrlContent($url);
+		if ($xml == false) {
```
Comment (LukasReschke) [inline lib/private/ocsclient.php]: `===`

File: lib/private/ocsclient.php [hunk 3]
```diff
@@ -21,19 +21,24 @@
...
 	/**
 	 * Returns whether the AppStore is enabled (i.e. because the AppStore is disabled for EE)
+	 *
 	 * @return bool
 	 */
-	protected static function isAppstoreEnabled() {
-		if(OC::$server->getConfig()->getSystemValue('appstoreenabled', true) === false OR OC_Util::getEditionString() !== '') {
+	public static function isAppStoreEnabled() {
+		if (\OC::$server->getConfig()->getSystemValue('appstoreenabled', true) === false ||
+			\OC_Util::getEditionString() !== ''
```
Comment (LukasReschke) [inline lib/private/ocsclient.php]: Guess this is unrequired anyways since AppStore is now enabled for EE?

**Comment by LukasReschke** [comment]
Besides my nitpicks: Nice stuff :rocket:

File: lib/private/ocsclient.php (hunk 3 above)
Comment (DeepDiver1975) [inline lib/private/ocsclient.php]: > Guess this is unrequired anyways since AppStore is now enabled for EE?

@karlitschek please comment - THX

**Comment by DeepDiver1975** [comment]
@LukasReschke addressed - thx
@nickvergessen @th3fallen please test and review - thx

File: lib/private/ocsclient.php (hunk 3 above)
Comment (karlitschek) [inline lib/private/ocsclient.php]: correct. this can go

**Comment by jancborchardt** [comment]
@etiess @clarjon1 please test this as it should your problems with showing the »Activate« button showing for already installed apps.

**Comment by DeepDiver1975** [comment]
@LukasReschke @nickvergessen @th3fallen please test and review - thx

**Comment by etiess** [comment]
@jancborchardt Sorry but I'm not used to test a version which is not in master :( I tried to download the nightly version on owncloud.org but the modification doesn't seem to be included.
Then I tried to download and install the zip from Github (branch "installing-installed-apps-is-stupid") but then, I cannot access owncloud in my browser (I get this message: "Composer autoloader not found, unable to continue. Check the folder "3rdparty".").
If you still need I test it, could you explain me how to do that? Thank you!

**Comment by LukasReschke** [comment]
Works :+1:

**Comment by LukasReschke** [comment]
> Then I tried to download and install the zip from Github (branch "installing-installed-apps-is-stupid") but then, I cannot access owncloud in my browser (I get this message: "Composer autoloader not found, unable to continue. Check the folder "3rdparty".").
> If you still need I test it, could you explain me how to do that? Thank you!

The easiest way is to get Git running as that allows you also to test further patches in the future. For that you'd then `git clone --recursive https://github.com/owncloud/core` and then `git checkout  installing-installed-apps-is-stupid` also from time to time go to the 3rdparty folder and do `git pull` to ensure that you're on the newest 3rdparty branch. (though there might be an easier command to do that @PVince81?)

If you want to test it just via the GitHub Web GUI you need to download this ZIP as well as https://github.com/owncloud/3rdparty and put all files in that folder into the `3rdparty` folder of the ownCloud installation. (it should be empty in the current state)

**Comment by PVince81** [comment]
`git submodule update` should do in any case

**Comment by etiess** [comment]
Thanks @LukasReschke and @PVince81 !
Tested, but it doesn't work on my installation: 
capture

**Comment by DeepDiver1975** [comment]
@etiess Thanks for testing! How did you install the already enabled Contacts app? THX

**Comment by etiess** [comment]
@DeepDiver1975 
On a fresh installation of OC downloaded via the web GUI of Github ("installing-installed-apps-is-stupid" branch + 3rdparty), I went to "PIM" and clicked on Activate fo the contact app. By the way, it is still not marked as Activated in PIM either.
capture

**Comment by DeepDiver1975** [comment]
@etiess Let me double check this quickly - THX

**Comment by DeepDiver1975** [comment]
@karlitschek @georgehrke 

There is something broken with the ocsid of calendar (contacts as well ???)

Id on the appstore: 168707 https://apps.owncloud.com/content/show.php/Calendar+8.0?content=168707
Id in info.xml: 166043 https://github.com/owncloud/calendar/blob/master/appinfo/info.xml#L20

If we cannot rely on the ocsid to match up - this will all not work.

**Comment by DeepDiver1975** [comment]
Ah - there is another calendar with the matching ocsid:
https://apps.owncloud.com/content/show.php/Calendar?content=166043

This is kind of :hankey:

**Comment by georgehrke** [comment]
> If we cannot rely on the ocsid to match up - this will all not work.

For some reason we published Calendar 8.0 and didn't use the (empty) calendar app that was already on apps.owncloud.com. The ocsid in the info.xml is the one of (empty) calendar app. Calendar 8.0 has a new id.
This should be fixed by replacing the old ocsid with the one of Calendar 8.0.

**Comment by DeepDiver1975** [comment]
> This should be fixed by replacing the old ocsid with the one of Calendar 8.0.

Please open a PR to fix this if time permits - THX

**Comment by DeepDiver1975** [comment]
Issue with calendar is unrelated to this PR -> merging

**Comment by DeepDiver1975** [comment]
> Please open a PR to fix this if time permits - THX

@georgehrke status on this? THX

**Comment by georgehrke** [comment]
> Please open a PR to fix this if time permits - THX

I don't see the need for a PR. We just need to update the calendar app on the App Store. Need to back port a few fixes and want to fix a few bugs, then publish a new version over the weekend.

**Comment by DeepDiver1975** [comment]
> I don't see the need for a PR. We just need to update the calendar app on the App Store. Need to back port a few fixes and want to fix a few bugs, then publish a new version over the weekend.

How ever you want to work in the calendar repo - I don't care much - as long as there will be an upload to the appstore which fixes the ocsis :wink: 

We also should remove/hide the old calendar entry @karlitschek @jancborchardt

**Comment by georgehrke** [comment]
> We also should remove/hide

It's disabled, so it doesn't show up in the app management, but still on apps.owncloud.com.
@karlitschek Is there a way to hide apps or can we safely delete it?

**Comment by georgehrke** [comment]
@DeepDiver1975 I thought you were talking about submitting a PR to core :smile:

**Comment by DeepDiver1975** [comment]
hehe - no :wink:

**Comment by karlitschek** [comment]
@georgehrke You can delete an app if it´s no longer needed. No problem.

**Comment by PVince81** [comment]
Follow up / related: https://github.com/owncloud/core/pull/14719 (stable8 PR).

**Pull request merged**